syntax = "proto3";
package xray;
import "google/protobuf/empty.proto";

message CertificateRequest {
    string certificate = 1;
    string server_id = 2;
}

message ClientIdRequest {
    string name = 1;
}

message ClientIdentifierResponse {
    string uuid = 1;
    string shortids = 2;
}

service XrayClientsService {
    rpc InstallCertificate (CertificateRequest) returns (google.protobuf.Empty);
    rpc AddClient(ClientIdRequest) returns (ClientIdentifierResponse);
    rpc DeleteClient(ClientIdRequest) returns (google.protobuf.Empty);
}

